### ISSUE TITLE
MarkDown link contents lost on save/document change

### ISSUE THREAD

**Issue by TomasHubelbauer** [typora/typora-issues#4591]
**Describe the bug**

I am facing a weird issue where when using a MarkDown link to another section within the document (`[text](#slug)`), in case the link URL contains backticks, those will survive a save, but will get lost when changing to a different document and back. Check the below sample MarkDown document.

Typora supports navigating to interior sections of the same document using the `#slug` link target and I have noticed it works both with hyphenated version (`#section-test-lol`) as well as verbatim version (``##Section `TEST` LOL``). I use the latter, because I can just copy-paste the section header text and do not have to hyphenate by hand, since Typora does not have auto-complete for internal links yet: #2663.

**To Reproduce**
1. Create a new document and paste the below code into it
2. Save and Ctrl+click the link to see that it works
3. Switch to a different document and back
4. Click into the link or enter source mode and see the link URL has changed to `#Section  LOL`
5. Ctrl+click the link again to see that it is broken

**Expected behavior**
The link target should not get broken.

**Sample Markdown File**
```markdown
# Test

## Section `TEST` LOL

Hi

## Section with a link

[Link to the `TEST` section](#Section `TEST` LOL)
```

**Desktop**
 - OS: Windows 10 Enterprise

**Typora Version**
0.11.2 beta

**Comment by abnerlee** [comment]
should be fixed in 0.11.5

**Comment by TomasHubelbauer** [comment]
Confirmed fixed, thank you.
